cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -531,6 +534,9 @@ protected void configure() {
         handlerBinder.addBinding().to(CredentialProviderHttpHandler.class);
         handlerBinder.addBinding().to(CredentialProviderHttpHandlerInternal.class);
         handlerBinder.addBinding().to(OperationHttpHandler.class);
+        handlerBinder.addBinding().to(MetadataHandler.class);
+        handlerBinder.addBinding().to(StoreHandler.class);
+        handlerBinder.addBinding().to(FetchHandler.class);
```

Comment:
similar comment for handlers should also not be added if `messagingService` is enabled.